nip47/controllers/get_budget_controller.go

```diff
@@ -0,0 +1,51 @@
+package controllers
+
+import (
+	"context"
+	"github.com/getAlby/hub/db/queries"
+	"github.com/nbd-wtf/go-nostr"
+
+	"github.com/getAlby/hub/db"
+	"github.com/getAlby/hub/logger"
+	"github.com/getAlby/hub/nip47/models"
+	"github.com/sirupsen/logrus"
+)
+
+type getBudgetResponse struct {
+	UsedBudget    uint64 `json:"used_budget"`
+	TotalBudget   uint64 `json:"total_budget"`
+	RenewsAt      uint64 `json:"renews_at"`
+	RenewalPeriod string `json:"renewal_period"`
+}
+
+func (controller *nip47Controller) HandleGetBudgetEvent(ctx context.Context, nip47Request *models.Request, requestEventId uint, app *db.App, publishResponse publishFunc) {
+
+	logger.Logger.WithFields(logrus.Fields{
+		"request_event_id": requestEventId,
+	}).Debug("Getting budget")
+
+	appPermission := db.AppPermission{}
+	controller.db.Where("app_id = ? AND scope = ?", app.ID, models.PAY_INVOICE_METHOD).First(&appPermission)
+
+	maxAmount := appPermission.MaxAmountSat
+	if maxAmount == 0 {
+		publishResponse(&models.Response{
```

Comment:
We actually did not catch this in your [proposal](https://github.com/nostr-protocol/nips/pull/1504). I am not sure an empty result makes sense. Maybe the connection should not support this method and return an error instead if the app tries to call this method when a budget is not set?

Reply:
Hmm that's a good alternative too. Maybe we should discuss on the NIP PR. There are some tradeoffs to think through for sure.